response_operations_ui/views/case.py

```diff
@@ -64,12 +64,17 @@ def get_response_statuses(ru_ref, error=None):
     case_id = get_case_by_case_group_id(case_group["id"]).get("id")
     is_complete = case_group_status in COMPLETE_STATE
     completed_timestamp = get_timestamp_for_completed_case_event(case_id) if is_complete else None
+    case_events = get_case_events_by_case_id(case_id=case_id)
 
     if case_group_status == "COMPLETE":
-        case_events = get_case_events_by_case_id(case_id=case_id)
         case_event = get_case_event_for_seft_or_eq(case_events)
         completed_respondent = get_user_from_case_events(case_event)
 
+    # Below code was added to remove duplicated and not needed 'Not started' radio buttons when the case is set to
+    # 'NOTSTARTED'
+    if case_group_status == "NOTSTARTED":
```

Comment:
We should try to absorb this into initial allowed status, so we don't have to step through the results again, it should also be explicit rather than removing anything that has a value of not started. I have had a quick play and come up with this, I have done a very quick refactor as well, because we twice checked to see if the case_group is complete and have a redundant case events call if it is

```
def allowed_statuses(case_group_status):
    return {"COMPLETEDBYPHONE", "NOLONGERREQUIRED"} if case_group_status == "NOTSTARTED" else {"NOTSTARTED"}
```

```
    case_groups = case_controller.get_case_groups_by_business_party_id(reporting_unit["id"])
    case_group = case_controller.get_case_group_by_collection_exercise(case_groups, exercise["id"])
    case_group_status = case_group["caseGroupStatus"]
    case_id = get_case_by_case_group_id(case_group["id"]).get("id")
    is_case_complete = case_group_status in COMPLETE_STATE
    completed_timestamp = get_timestamp_for_completed_case_event(case_id) if is_case_complete else None
    
    if is_case_complete:
        case_events = get_case_events_by_case_id(case_id=case_id)
        case_event = get_case_event_for_seft_or_eq(case_events)
        completed_respondent = get_user_from_case_events(case_event)

    case_statuses_allowed = case_controller.allowed_statuses(case_group_status)
    available_statuses = {
        event: map_ce_response_status(status)
        for event, status in statuses.items()
        if status in case_statuses_allowed
```

Reply:
You might have to tweak this as I am think we might also have INPROGRESS as a state

Reply:
Agreed, I was going to make a comment similar to this.  We could/should maybe go a bit further and for all the states that we allow changes to (notstarted, nolongerrequired and completedbyphone I think) we get the list of all the allowed transitions, filter out all the ones which just go back to themselves (like NOTSTARTED -> action -> NOTSTARTED) then display the list.

The way the PR is now probably works as expected but is still a bit hard to follow I think.